internal/core/metadata/uom/bootstrap.go

```diff
@@ -36,7 +38,14 @@ func BootstrapHandler(_ context.Context, _ *sync.WaitGroup, _ startup.Timer, dic
 		return true
 	}
 
-	contents, err := os.ReadFile(filepath)
+	requestTimeout, err := time.ParseDuration(config.Service.RequestTimeout)
+	if err != nil {
+		lc.Debug("Failed to parse Service.RequestTimeout configuration value: %v, using default value", err)
+		requestTimeout = 15 * time.Second
+	}
+
+	secretProvider := container.SecretProviderFrom(dic.Get)
+	contents, err := file.Load(filepath, requestTimeout, secretProvider)
 	if err != nil {
 		lc.Errorf("could not load unit of measure configuration file (%s): %s", filepath, err.Error())
```

Comment:
This could leak credential if in filepath. Error should already have clean path in it.
```suggestion
		lc.Errorf("could not load unit of measure configuration file: %s", err.Error())
```